doc/syntax.md

```diff
@@ -2,42 +2,323 @@
+* `boolean`: corresponds to the `bool` data type in C++. Its literals only are:
+`true`, `false`
+* `string`: String data type. Its literals are: "Bob", ""
+* `void`: corresponds to the `void` data type in C++. No literals are possible.
+It can be used to identify a function without the return value.
+
+#### Physical units
+A physical unit in NESTML can be either a simple physical unit or a complex
+physical unit. A simple physical unit is composed of an optional magnitude prefix
+and the name of the unit.
+
+The following table present seven base units, which can be used to specify any
+physical unit. This idea is based on the Le Système internationale d’unités.
+
+|Quantity | SI Symbol | SI Name | NESTML Name|
+|-|-|-| - |
+| length | L | metre  | m |
+| mass | M | kilogram | kg |
+| time | T | second | s |
+| electric current | I | ampere | A |
+| temperature | $\Theta$ | kelvin | K |
+| amount of substance | N | mole | mol |
+| luminous intensity | J | candela | cd |
+
+All remaining physical units are called derived units. Theoretically, any physical
+unit can be expressed as a combination of these seven
+units. In order to support convenience in modeling, NESTML provides following
+derived units as built-in data types:
+
+| | | | |
+|-|-|-|-|
+| Bq | C | F | Gy |
+| Hz | J | N | Ohm |
+| Pa | S | Sv | TH |
+| V | W  | Wb | kat | |
+
+Units can have at most one of the following magnitude prefixes:
+
+|Factor | SI Name | NESTML prefix | Factor | SI Name | NESTML prefix
+|-|-|-|-|-|-|
+|$10^{-1}$ | deci | d | $10^1$ | deca | da |
+|$10^{-2}$ | centi | c | $10^2$ | hecto | h |
+|$10^{-3}$ | milli | m | $10^3$ | kilo | k |
+|10^{-6}$ | micro | mu | $10^6$ | mega | M |
+|$10^{-9}$ | nano | n | $10^9$ | giga | G |
+|10^{-12}$ | pico | p | $10^{12}$ | tera | T |
+|$10^{-15}$ | femto | f | $10^{15}$ | peta | P |
+|$10^{-18}$ | atto | a | $10^{18}$ | exa | E |
+|$10^{-21}$ | zepto | z | $10^{21}$ | zetta | Z |
+|$10^{-24}$ | yocto | y | $10^{24}$ | yotta | Y |
+
+Finally, simple physical units can be combined to complex units. For this,
+parenthesis(`()`) , multiplication (\*)-, division(\)- and power(\*\*)-operators
+can be used.
+```
+mV*mV*nS**2/(mS*pA)
+```
+There are special cases for the expressions of the form `physical_unit ** -1`.
+It can be expressed as `1\physical_unit`
+```
+(ms*mV)**-1 is equivalent to 1/(ms*mV)
+```
+
+#### Physical unit literals
+Simple unit literals are composed of a number with the type name (with or without a whitespace).
+Complex unit literals are composed of a number and type inside of square brackets.
+Also simple units can be put inside of square brackets.  
+
+```
+<number> <unit_type>
+<number> [<unit_type>]
+
+a mV = 1mV
+aa mV = 1[mV]
+b mV/ms = 1[mV/ms]
+```
+#### Type and unit checks
+NESTML checks type correctness of all expressions. For assignments, declarations
+with an initialization and function calls the type conformity is checked also.
```

Comment:
Complementary, I will extend my tutorial (not reference manual) with a list of context conditions (about 30). After that we can copy and paste them into manual. They, actually define the type system of nestml.